gui/default/syncthing/core/syncthingController.js

```diff
@@ -1178,8 +1178,24 @@ angular.module('syncthing.core')
                 $scope.tmpOptions.upgrades = "candidate";
             }
             $scope.tmpGUI = angular.copy($scope.config.gui);
-            $('#settings').modal().one('hidden.bs.modal', function () {
+            $scope.tmpRemoteIgnoredDevices = angular.copy($scope.config.remoteIgnoredDevices);
+            $scope.tmpRemoteIgnoredFolders = angular.copy($scope.config.remoteIgnoredFolders);
+            var settingsModal = $('#settings').modal();
+            settingsModal.one('hidden.bs.modal', function () {
+                $('.nav-tabs a[href="#settings-general"]').tab('show');
                 window.location.hash = "";
+                settingsModal.off('hide.bs.modal');
+            }).on('hide.bs.modal', function (e) {
+                if ($scope.settingsModified()) {
+                    $("#discard-changes-confirmation").modal().one('hidden.bs.modal', function() {
+                        if (!$scope.settingsModified()) {
```

Comment:
Why is there an "outer" and "inner" check for `settingsModified`?  
This branch presumably runs after `saveSettings` when the user hits `Save` in the settings modal, right?

Reply:
Essentially if they are no longer modified (been overwritten with defaults by pressing yes in the question modal), we close the parent window, otherwise, the user pressed no, and we leave the parent dialog visible.